modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go

```diff
@@ -960,6 +960,154 @@ func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeAck() {
 	}
 }
 
+func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeOpen() {
+	var (
+		path                *ibctesting.Path
+		isNilApp            bool
+		counterpartyVersion string
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
```

Comment:
Suppose there will never an `expError` because the func can only panic in one case. 

It should be possible to set some mock app or middleware as the app which doesn't implement the interface. Maybe could reuse something from https://github.com/cosmos/ibc-go/pull/5406 with `expPanics`, where instead of using `nil` we use a mock that doesn't implement the interface:

```
if expPanics {
    cbs = controller.NewIBCMiddleware(ibcmock.NewBlockUpgradeMiddleware(), suite.chainA.GetSimApp().ICAControllerKeeper)
}
```

Happy to do it in a follow up if we go with the mock upgrade middleware